include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,525 @@
+        }
+        // case 1: complete gap is deleted
+        else if (gap_len == pos2 - pos1)
+        {
+            it = anchors.erase(it);
+        }
+        // case 2: gap to be deleted in tail or larger than 1 (equiv. to shift tail left, i.e. pos remains unchanged)
+        else
+        {
+            gap_t gap{(*it).first, (*it).second - pos2 + pos1};
+            it = anchors.erase(it);
+            it = anchors.insert(it, gap); // amortized constant because of hint
+            ++it; // update node after the current
+        }
+
+        // post-processing: forward update of succeeding gaps
+        update(it, pos2 - pos1);
+
+        return first;
+    }
+
+    /*!\name Iterators
+     * \{
+     */
+    /*!\brief Returns an iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() noexcept
+    {
+        return iterator{*this};
+    }
+
+    //!\copydoc begin()
+    const_iterator begin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    //!\copydoc begin()
+    const_iterator cbegin() const noexcept
+    {
+        return const_iterator{*this};
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the decorator.
+     * \returns Iterator to the behind last element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() noexcept
+    {
+        return iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator end() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+
+    //!\copydoc end()
+    const_iterator cend() const noexcept
+    {
+        return const_iterator{*this, size()};
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the i-th element as a reference.
+     * \param i     The element to retrieve.
+     * \returns     A reference of the gapped alphabet type.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * Throws std::out_of_range exception if \p i is out of range.
+     */
+    reference at(size_type const i)
+    {
+        //TODO add SEQAN_UNLIKELY
+        if (i >= size())
+            throw std::out_of_range{"Trying to access element behind the last in gap_decorator."};
+        return (*this)[i];
+    }
+
+    //!\copydoc at()
+    const_reference at(size_type const i) const
+    {
+        //TODO add SEQAN_UNLIKELY
+        if (i >= size())
```

Comment:
```suggestion
        if (i >= size()) // [[unlikely]]
```
and remove comment in line above